src/m365/cli/commands/app/app-reconsent.ts

```diff
@@ -0,0 +1,127 @@
+import GraphCommand from '../../../base/GraphCommand.js';
+import commands from '../../commands.js';
+import { Logger } from '../../../../cli/Logger.js';
+import auth from '../../../../Auth.js';
+import { odata } from '../../../../utils/odata.js';
+import { Application, RequiredResourceAccess, ServicePrincipal } from '@microsoft/microsoft-graph-types';
+import config from '../../../../config.js';
+import { urlUtil } from '../../../../utils/urlUtil.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { cli } from '../../../../cli/cli.js';
+import { settingsNames } from '../../../../settingsNames.js';
+import { browserUtil } from '../../../../utils/browserUtil.js';
+
+class CliAppReconsentCommand extends GraphCommand {
+  public get name(): string {
+    return commands.APP_RECONSENT;
+  }
+
+  public get description(): string {
+    return 'Reconsent all permission scopes used in CLI for Microsoft 365';
+  }
+
+  public async commandAction(logger: Logger): Promise<void> {
+    try {
+      const appId = auth.connection.appId!;
+      if (this.verbose) {
+        await logger.logToStderr(`Adding all missing permission scopes used in CLI for Microsoft 365 to application with ID '${appId}'...`);
+      }
+
+      const application = await this.getApplicationByAppId(appId);
+      await this.addCliAppScopes(logger, application.requiredResourceAccess!);
+      await this.updateAppScopes(logger, application.id!, application.requiredResourceAccess!);
+
+      const consentUrl = `https://login.microsoftonline.com/${auth.connection.tenant}/adminconsent?client_id=${appId}`;
+      await logger.log(`To consent to the new scopes for your Microsoft Entra application registration, please navigate to the following URL: ${consentUrl}`);
+
+      if (cli.getSettingWithDefaultValue(settingsNames.autoOpenLinksInBrowser, false)) {
+        await browserUtil.open(consentUrl);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getApplicationByAppId(appId: string): Promise<Application> {
+    const apps = await odata.getAllItems<Application>(`${this.resource}/v1.0/applications?$filter=appId eq '${appId}'&$select=requiredResourceAccess,id`);
+    return apps[0];
+  }
+
+  private async addCliAppScopes(logger: Logger, appScopes: RequiredResourceAccess[]): Promise<void> {
+    const allCliScopes = config.allScopes;
+    const servicePrincipals = await odata.getAllItems<ServicePrincipal>(`${this.resource}/v1.0/servicePrincipals?$select=displayName,appId,oauth2PermissionScopes,servicePrincipalNames`);
```

Comment:
Let's change this to use our util `entraServicePrincipal.getServicePrincipalByAppId`